src/provider/utils.ts

```diff
@@ -49,7 +49,7 @@ export function getBlockIdentifier(blockIdentifier: BlockIdentifier): BlockIdent
   if (blockIdentifier === 'pending') {
     return { type: 'BLOCK_NUMBER', data: 'pending' };
   }
-  if (typeof blockIdentifier === 'number') {
+  if (typeof blockIdentifier === 'number' || typeof blockIdentifier === 'bigint') {
```

Comment:
Because there was no status code check from the fetchEndpoint method, the test case 
test(`getBlock(blockHash=undefined, blockNumber=${exampleBlockNumber})`, () => {
      return expect(provider.getBlock(exampleBlockNumber)).resolves.not.toThrow();
    });
passed 
but if you look at the logs, it resolves the blockNumber in this case to blockHash
i.e. http://127.0.0.1:5050/feeder_gateway/get_block?blockHash=0x55

Reply:
nice catch